mPowerSDK/TasksAndSteps/MedicationTracker/APHMedicationTrackerTask.m

```diff
@@ -0,0 +1,717 @@
+        return (self.subTask != nil) && self.dataStore.shouldIncludeMomentInDayStep;
+    }
+    return [self shouldIncludeMedicationTrackingSteps];
+}
+
+- (BOOL)shouldIncludeMedicationTrackingSteps {
+    if (self.subTask != nil) {
+        // If there is a subtask then include only if the question has no answer and hasn't been skipped
+        return self.dataStore.hasChanges || !self.dataStore.hasSelectedMedicationOrSkipped;
+    }
+    return YES;
+}
+
+- (BOOL)shouldUpdateAndIncludeFrequencyStep:(ORKStep*)step {
+    
+    ORKFormStep *formStep = (ORKFormStep *)step;
+    if (![formStep isKindOfClass:[ORKFormStep class]]) {
+        NSAssert(NO, @"Medication frequency step is not of expected class");
+        return NO;
+    }
+    
+    // Add the list of medications
+    NSMutableArray *items = [NSMutableArray new];
+    for (APHMedication *med in self.dataStore.selectedMedications) {
+        if (!med.injection) {
+            ORKAnswerFormat *answerFormat = [[ORKScaleAnswerFormat alloc] initWithMaximumValue:12 minimumValue:1 defaultValue:0 step:1];
+            ORKFormItem  *item = [[ORKFormItem alloc] initWithIdentifier:med.identifier
+                                                                    text:med.text
+                                                            answerFormat:answerFormat];
+            [items addObject:item];
+        }
+    }
+    
+    formStep.formItems = items;
+    
+    return (items.count > 0);
+}
+
+- (BOOL)shouldUpdateAndIncludeMomentInDayStep:(ORKStep *)step {
+    
+    ORKFormStep *formStep = (ORKFormStep *)step;
+    if (![formStep isKindOfClass:[ORKFormStep class]]) {
+        NSAssert(NO, @"Medication frequency step is not of expected class");
+        return NO;
+    }
+    
+    if ((self.subTask == nil) || !self.dataStore.shouldIncludeMomentInDayStep) {
+        return NO;
+    }
+
+    // Get the medication list
+    NSArray *medList = self.dataStore.trackedMedications;
+    
+    // Build list of meds
+    NSString *itemTextFormat = NSLocalizedStringWithDefaultValue(@"APH_MOMENT_IN_DAY_QUESTION", nil, APHLocaleBundle(), @"When was the last time you took your %@?", @"Prompt for timing of medication in pre-activity medication timing survey where %@ is a list of medications (For example, 'Levodopa or Rytary')");
+    NSString *orWord = NSLocalizedStringWithDefaultValue(@"APH_OR_FORMAT", nil, APHLocaleBundle(), @"or", @"Format of a list with two items using the OR key word.");
+    NSString *listDelimiter = NSLocalizedStringWithDefaultValue(@"APH_LIST_FORMAT_DELIMITER", nil, APHLocaleBundle(), @",", @"Delimiter for a list of more than 3 items. (For example, 'Levodopa, Simet or Rytary')");
+    
+    NSMutableString *listText = [NSMutableString new];
+
+    [medList enumerateObjectsUsingBlock:^(NSString * _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop __unused) {
+        if (medList.count > 1) {
+            if (idx+1 == medList.count) {
```

Comment:
Not sure how well this approach to lists will stand up in other languages, but I guess we can rethink it later if it turns out not to